frontends/web/src/routes/device/bitbox02/gotostartupsettings.tsx

```diff
@@ -14,55 +15,40 @@
  * limitations under the License.
  */
 
-import { Component } from 'react';
-import { translate, TranslateProps } from '../../../decorators/translate';
+import { useState } from 'react';
 import { apiPost } from '../../../utils/request';
 import { SettingsButton } from '../../../components/settingsButton/settingsButton';
 import { WaitDialog } from '../../../components/wait-dialog/wait-dialog';
+import { useTranslation } from 'react-i18next';
 
 interface GotoStartupSettingsProps {
     apiPrefix: string;
 }
 
-type Props = GotoStartupSettingsProps & TranslateProps;
+export const GotoStartupSettings = ({ apiPrefix }: GotoStartupSettingsProps) => {
+  const [isConfirming, setIsConfirming] = useState(false);
+  const { t } = useTranslation();
 
-interface State {
-    isConfirming: boolean;
-}
-
-class GotoStartupSettings extends Component<Props, State> {
-  public readonly state: State = {
-    isConfirming: false,
-  };
-
-  private gotoStartupSettings = (): void => {
-    this.setState({ isConfirming: true });
-    apiPost(this.props.apiPrefix + '/goto-startup-settings').then(() => {
-      this.setState({ isConfirming: false });
-    });
+  const gotoStartupSettings = async () => {
+    setIsConfirming(true);
+    await apiPost(`${apiPrefix}/goto-startup-settings`).catch(console.error);
```

Comment:
Not sure if you want to do it in this PR, but in the past I have replaced apiPrefix with deviceID, I think it is better.

```suggestion
    await apiPost(`devices/bitbox02/${deviceID}/goto-startup-settings`).catch(console.error);
```

Reply:
Yeah, lets do that to maintain consistency. Thanks for the suggestion!:)